quizapp/app/src/main/java/pl/edu/agh/toik/quizapp/QuizActivity.kt

```diff
@@ -21,12 +21,17 @@ class QuizActivity : AppCompatActivity() {
         setContentView(R.layout.activity_quiz)
         val quantity = intent.getIntExtra(EXTRA_QUANTITY, 5)
         val difficulty = Difficulty.valueOf(intent.getStringExtra(EXTRA_DIFFICULTY)!!)
-        val category = intent.getLongExtra(EXTRA_CATEGORY, 0)
-        val questions = QuizApi().getQuizQuestions(quantity, difficulty, category)
+        val categoryId = intent.getLongExtra(EXTRA_CATEGORY, 0)
+        val questions = getQuestions(quantity, difficulty, categoryId)
+        checkQuestionsPresence(questions)
```

Comment:
When I tap outside dialog or select _Back_ button, the empty QuizActivity is still there (tapping _Next_ redirect to empty StatisticsActivity as well)